mlir/lib/Dialect/Affine/IR/ValueBoundsOpInterfaceImpl.cpp

```diff
@@ -88,6 +109,27 @@ struct AffineMaxOpInterface
           expr.replaceDimsAndSymbols(dimReplacements, symReplacements);
       cstr.bound(value) >= bound;
     }
+    // Get all constant upper bounds, choose maximum, and set upper bound to it.
+    MLIRContext *ctx = op->getContext();
```

Comment:
This piece of code looks very similar to the one above. Maybe some functionality can be put into a shared static function. Not sure if it's really possible / practical...